cpp/bridges_module/algorithm/bridges.cpp

```diff
@@ -0,0 +1,53 @@
+#include <algorithm>
+
+#include "bridges.hpp"
+
+bridges_util::NodeState::NodeState(std::size_t number_of_nodes) {
+  visited.resize(number_of_nodes, false);
+  discovery.resize(number_of_nodes, 0);
+  low_link.resize(number_of_nodes, 0);
+  counter = 0;
+}
+
+void bridges_util::NodeState::Update(std::uint64_t node_id) {
+  counter++;
+  visited[node_id] = true;
+  discovery[node_id] = counter;
+  low_link[node_id] = counter;
+}
+
+void bridges_util::BridgeDfs(std::uint64_t node_id, std::uint64_t parent_id, bridges_util::NodeState *state,
+                             std::vector<mg_graph::Edge<>> *bridges, const mg_graph::GraphView<> &graph) {
+  state->Update(node_id);
+
+  for (const auto &neigh : graph.Neighbours(node_id)) {
+    auto next_id = neigh.node_id;
+    if (state->visited[next_id]) {
+      if (next_id != parent_id) {
+        state->low_link[node_id] = std::min(state->low_link[node_id], state->discovery[next_id]);
+      }
+      continue;
+    }
+
+    bridges_util::BridgeDfs(next_id, node_id, state, bridges, graph);
+    state->low_link[node_id] = std::min(state->low_link[node_id], state->low_link[next_id]);
+
+    const auto &edge = graph.GetEdge(neigh.edge_id);
+    if (state->low_link[next_id] > state->discovery[node_id]) {
+      if (graph.GetEdgesBetweenNodes(edge.from, edge.to).size() == 1) bridges->push_back(edge);
+    }
+  }
+}
```

Comment:
Since all these functions are defined in `bridefs_util` namespace you can write it like this:
```suggestion
namespace bridges_util {
  NodeState::NodeState(std::size_t number_of_nodes) {
...
}
```
 in order to avoid verbose definitions like this`bridges_util::NodeState::Update`